File: Atcoder_#352/b_typing.rb
s = gets.chomp.split('')
t = gets.chomp.split('')

# 正しい文字列の格納先を初期化
true_chars = []
# 再走査する際に利用する
last_char = 0

# sを左から抽出
s.each do |s_char|
  # tを左からsと比較
  # tは次の文字の走査の際には、前回のindexからスタート
  (last_char...t.size).each do |index|
    if s_char == t[index]
      true_chars << index + 1
      last_char = index + 1
      break
    end
  end
end

puts true_chars.join(" ")

File: Atcoder_#352/b_typing.js
const readline = require('readline');

// 標準入力を読み取るためのインターフェースを作成
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let inputLines = [];

// 標準入力を読み取るイベントリスナー
rl.on('line', (line) => {
  inputLines.push(line);
}).on('close', () => {
  // 入力データを処理する関数を呼び出し
  processInput(inputLines);
});

function processInput(input) {
  const s = input[0].split('');
  const t = input[1].split('');

  let trueChars = [];
  let lastChar = 0;

  s.forEach((sChar) => {
    for (let index = lastChar; index < t.length; index++) {
      if (sChar === t[index]) {
        trueChars.push(index + 1);
        lastChar = index + 1;
        break;
      }
    }
  });

  console.log(trueChars.join(' '));
}
